docs/workflows/transitions/preprocess/preprocess/make_predictions.py

```diff
@@ -5,21 +5,42 @@
 import base64
 
 
+logging.getLogger().setLevel(logging.INFO)
+
+
 def required_labels(field_config):
     return set([label for label in field_config if field_config[label].get('required', True)])
 
 
+def filter_optional_fields(predictions, field_config):
+    def predicate(p):
+        conf_threshold = field_config[p['label']]['confidenceLevels']['low']
+        return p['label'] in required_labels(field_config) or conf_threshold < p['confidence']
+    
+    return list(filter(predicate, predictions))
+
+
+def filter_by_top1(predictions):
+    labels = set(map(lambda p: p['label'], predictions))
+    
+    def top1(label):
+        preds = filter(lambda f: f['label'] == label, predictions)
+        return max(preds, key=lambda p: p['confidence'])
+
+    return [top1(label) for label in labels]
```

Comment:
What happens if we have no predictions for a certain label, looks like `max(...)` will throw?